storage_service/administration/forms.py

```diff
@@ -222,8 +222,17 @@ class UserChangeForm(auth.forms.UserChangeForm):
     """ Modifys an existing user.  Inherits from django's UserChangeForm. """
 
     def __init__(self, *args, **kwargs):
+        self.current_user = kwargs.pop("current_user", None).username
         super(UserChangeForm, self).__init__(*args, **kwargs)
+        currently_editing = getattr(kwargs["instance"], "username")
+        superusers = auth.get_user_model().objects.filter(is_superuser=True)
         self.fields["is_superuser"].label = _("Administrator?")
+        if len(superusers) >= 1:
+            if (
+                not getattr(kwargs["instance"], "is_superuser")
+                and self.current_user == currently_editing
+            ):
+                del self.fields["is_superuser"]
```

Comment:
I think this is the right approach; the form is the right place to make this change. It's kind of too bad there isn't a better solution for this in Django but none of the permissions logic gets as granular as form fields. 

I would simplify it though so that only superusers can change the attribute, they can change it for anyone, and we don't care if there are any existing superusers or not. 

something like:
```
current_user = kwargs.pop("current_user", None)
...
if not (current_user and current_user.is_superuser):
    del self.fields["is_superuser"]
```

I'm guessing the check for existing superusers is to prevent being locked out of the system? Django provides a management command (`createsuperuser`) for that case. It's not very user friendly but it seems like it would be ok in the event someone locks out all accounts. Alternatively I think more secure approach would be to disable the superuser field on the last super user. In that case we can disable the form field:

```
superusers = auth.get_user_model().objects.filter(is_superuser=True)
...
if not (current_user and current_user.is_superuser):
    del self.fields["is_superuser"]
elif superusers.count() == 1:
    self.fields['is_superuser'].widget.attrs['readonly']
```